mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java

```diff
@@ -169,23 +180,70 @@ GroupRegistration start() {
     }
 
     private void subscribeWorkQueue() {
-        receiverSubscriber = Optional.of(receiver.consumeAutoAck(queueName.asString())
+        receiverSubscriber = Optional.of(receiver.consumeManualAck(queueName.asString())
             .subscribeOn(Schedulers.parallel())
-            .map(Delivery::getBody)
-            .filter(Objects::nonNull)
-            .map(eventInBytes -> new String(eventInBytes, StandardCharsets.UTF_8))
-            .map(eventSerializer::fromJson)
-            .map(JsResult::get)
+            .filter(delivery -> Objects.nonNull(delivery.getBody()))
+            .flatMap(this::deliver)
             .subscribeOn(Schedulers.elastic())
-            .subscribe(event -> deliverEvent(mailboxListener, event)));
+            .subscribe());
     }
 
-    private void deliverEvent(MailboxListener mailboxListener, Event event) {
-        try {
-            mailboxListener.event(event);
-        } catch (Exception e) {
-            LOGGER.error("Exception happens when handling event of user {}", event.getUser().asString(), e);
+    private Mono<Void> deliver(AcknowledgableDelivery acknowledgableDelivery) {
+        byte[] eventAsBytes = acknowledgableDelivery.getBody();
+        Event event = eventSerializer.fromJson(new String(eventAsBytes, StandardCharsets.UTF_8)).get();
+        int currentRetryCount = getRetryCount(acknowledgableDelivery);
+
+        return Mono.just(currentRetryCount)
+            .filter(retryCount -> retryCount <= MAX_RETRIES)
+            .delayElement(waitDelay(currentRetryCount))
+            .flatMap(any -> Mono.fromRunnable(() -> mailboxListener.event(event)))
+            .doOnError(throwable -> LOGGER.error("Exception happens when handling event of user {}", event.getUser().asString(), throwable))
+            .onErrorResume(throwable -> handleRetryOnError(event, eventAsBytes, currentRetryCount))
+            .doOnSuccess(success -> acknowledgableDelivery.ack())
+            .subscribeWith(MonoProcessor.create())
+            .then();
+    }
+
+    private Duration waitDelay(int retryCount) {
+        if (retryCount < 1) {
+            return Duration.ZERO;
         }
+        int exponentialFactor = Double.valueOf(Math.pow(2, retryCount)).intValue();
+        int minDelay = exponentialFactor * FIRST_BACKOFF_MS;
+        int maxDelay = Double.valueOf(minDelay + minDelay * DEFAULT_JITTER_FACTOR).intValue(); // maxDelay = 150% of minDelay
+
+        return Duration.ofMillis(randomBetween(minDelay, maxDelay));
+    }
+
+    private int randomBetween(int lowest, int highest) {
+        SecureRandom random = new SecureRandom();
```

Comment:
`SecureRandom` should be a static final constant

Reply:
What happens if lowest is higher than highest? Do we need a precondition on this?